drivers/can/stm32_can.c

```diff
@@ -690,14 +690,11 @@ static inline int can_stm32_attach(struct device *dev, void *response_ptr,
 	int filter_index_tmp = 0;
 	int filter_nr;
 
-	k_mutex_lock(&data->set_filter_mutex, K_FOREVER);
```

Comment:
curious why these mutex calls were removed? commit message does not mention why.

Reply:
I moved it to protect data->response_type